tests/bgp/test_bgp_suppress_fib.py

```diff
@@ -0,0 +1,588 @@
+    for route in IPV6_ROUTE_LIST:
+        check_route_install_status(duthost, route, vrf, IPV6_VER, check_point, action)
+
+
+def validate_route_propagate(duthost, tbinfo, vrf=DEFAULT, exist=True):
+    """
+    Verify ipv4 and ipv6 route propagate status
+    """
+    t2_vm = get_vm_name(tbinfo)
+    bgp_neighbor_v4, bgp_neighbor_v6 = get_bgp_neighbor_ip(duthost, t2_vm, vrf)
+    for route in IP_ROUTE_LIST:
+        validate_route_propagate_status(duthost, route, bgp_neighbor_v4, vrf, exist=exist)
+    for route in IPV6_ROUTE_LIST:
+        validate_route_propagate_status(duthost, route, bgp_neighbor_v6, vrf, ip_ver=IPV6_VER, exist=exist)
+
+
+def redistribute_static_route_to_bgp(duthost, redistribute=True):
+    """
+    Enable or disable redistribute static route to BGP
+    """
+    vtysh_cmd = "sudo vtysh"
+    config_terminal = " -c 'config'"
+    enter_bgp_mode = " -c 'router bgp'"
+    enter_address_family_ipv4 = " -c 'address-family ipv4'"
+    redistribute_static = " -c 'redistribute static'"
+    no_redistribute_static = " -c 'no redistribute static'"
+    if redistribute:
+        duthost.shell(vtysh_cmd + config_terminal + enter_bgp_mode + enter_address_family_ipv4 + redistribute_static)
+    else:
+        duthost.shell(vtysh_cmd + config_terminal + enter_bgp_mode + enter_address_family_ipv4 + no_redistribute_static)
+
+
+def remove_static_route_and_redistribute(duthost):
+    """
+    Remove static route and stop redistribute it to BGP
+    """
+    out = duthost.shell("show ip route {}".format(STATIC_ROUTE_PREFIX), verbose=False)['stdout']
+    if STATIC_ROUTE_PREFIX in out:
+        duthost.shell("sudo config route del prefix {}".format(STATIC_ROUTE_PREFIX))
+        redistribute_static_route_to_bgp(duthost, redistribute=False)
+
+
+@pytest.mark.parametrize("vrf_type", VRF_TYPES)
+def test_bgp_route_with_suppress(duthost, tbinfo, nbrhosts, ptfadapter, localhost, restore_bgp_suppress_fib,
+                                 get_exabgp_ptf_ports, vrf_type):
+    try:
+        if vrf_type == USER_DEFINED_VRF:
+            with allure.step("Configure user defined vrf"):
+                setup_vrf(duthost, nbrhosts, tbinfo)
+
+        with allure.step("Prepare needed parameters"):
+            router_mac = duthost.facts["router_mac"]
+            mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+            ptf_ip = tbinfo['ptf_ip']
+            exabgp_port, ptf_recv_port, exabgp_port_v6, ptf_recv_port_v6 = get_exabgp_ptf_ports
+            recv_port = {
+                4: ptf_recv_port,
+                6: ptf_recv_port_v6
+            }
+
+        with allure.step("Config bgp suppress-fib-pending function"):
+            config_bgp_suppress_fib(duthost)
+
+        with allure.step("Save configuration and execute one action randomly \
+        choosen from reboot/config reload/fast-reboot/warm-reboot"):
+            randomly_reboot(duthost, localhost)
```

Comment:
@echuawu what's the reason to use randomly reboot? Could we use reboot method as a parameter.

Reply:
Hi @StormLiangMS , it's purpose is to cover all the reboot types supported and keep time saving strategy.

Reply:
@echuawu 1. what's the reason to reboot/reload at the beginning of the test? 2. what's the reason to use random? can we fixed with reload? 3. to use parameterize, it would increase the cost of CI/CD, need to justify it.

Reply:
Hi @StormLiangMS , consider the test scope coverage, for nvidia, we need to cover all kinds of reload. So I propose it fix with config reload by default, and it also could config with a parameter to do a random reboot. What's your idea?

Reply:
@echuawu can I understand this test is to check if this feature can work after reload?

Reply:
@StormLiangMS , yes, make sure it could work after all kinds of reboot and reload.

Reply:
@StormLiangMS , code has been updated.